common/acpi.c

```diff
@@ -134,29 +134,12 @@ static rsdp_rev1_t *acpi_find_rsdp(void) {
     return NULL;
 }
 
-static unsigned acpi_table_map_pages(paddr_t pa, size_t len) {
-    unsigned offset = pa & ~PAGE_MASK;
-    unsigned num_pages = ((offset + len) / PAGE_SIZE) + 1;
-    mfn_t mfn = paddr_to_mfn(pa);
-
-    for (unsigned i = 0; i < num_pages; i++, mfn++) {
-        if (mfn_invalid(mfn)) {
-            panic("ACPI table at %p of length %lx has invalid MFN: %lx\n", _ptr(pa), len,
-                  mfn);
-        }
-
-        BUG_ON(!kmap_4k(mfn, L1_PROT));
-    }
-
-    return num_pages;
+static void *acpi_table_map_pages(paddr_t pa, size_t len) {
+    return mmap_range(pa, len, L1_PROT, MMAP_KERNEL);
```

Comment:
The old code handled non-page aligned `pa`, maybe `mmap_range()` should too?

Reply:
I think it still does so.

Reply:
You're right, I was misreading `mmap_range()`'s implementation.